src/metabase/email/messages.clj

```diff
@@ -246,6 +251,62 @@
+        monitoring (perms/application-perms-path :monitoring)
+        db-details (perms/feature-perms-path :details :yes database-id)
+        user-ids (->> {:select   [:pgm.user_id]
+                       :from     [[:permissions_group_membership :pgm]]
+                       :join     [[:permissions_group :pg] [:= :pgm.group_id :pg.id]]
+                       :where    [:and
+                                  [:exists {:select [1]
+                                            :from [[:permissions :p]]
+                                            :where [:and
+                                                    [:= :p.group_id :pg.id]
+                                                    [:= :p.object monitoring]]}]
+
+                                  [:exists {:select [1]
+                                            :from [[:permissions :p]]
+                                            :where [:and
+                                                    [:= :p.group_id :pg.id]
+                                                    [:= :p.object db-details]]}]]
+                       :group-by [:pgm.user_id]}
+                      db/query
+                      (mapv :user_id))
+        emails (db/select-field :email User {:where [:and
+                                                     [:= :is_active true]
+                                                     [:or
+                                                      [:= :is_superuser true]
+                                                      [:in :id user-ids]]]})
+        timezone (some-> database qp.timezone/results-timezone-id t/zone-id)
+        context {:database-name (:name database)
+                 :errors
+                 (for [persisted-info persisted-infos
+                       :let [card (:card persisted-info)
+                             collection (or (:collection card)
+                                            (collection/root-collection-with-ui-details nil))]]
+                   {:error (:error persisted-info)
+                    :card-id (:id card)
+                    :card-name (:name card)
+                    :collection-name (:name collection)
+                    ;; February 1, 2022, 3:10 PM
+                    :last-run-at (t/format "MMMM d, yyyy, h:mm a z" (t/zoned-date-time (:refresh_begin persisted-info) timezone))
+                    :last-run-trigger trigger
+                    :card-url (urls/card-url (:id card))
+                    :collection-url (urls/collection-url (:id collection))
+                    :caching-log-details-url (urls/tools-caching-details-url (:id persisted-info))})}
+        message-body (stencil/render-file "metabase/email/persisted-model-error"
+                                          (merge (common-context) context))]
+    (when (seq emails)
+      (email/send-message!
+        :subject      (trs "[{0}] Model cache refresh failed for {1}" (app-name-trs) (:name database))
+        :recipients   (vec emails)
```

Comment:
kinda nothing but what is the purpose of the `vec` call here?

Reply:
makes me nervous because its either unnecessary or `email/send-message!` is kinda weird and there be dragons but i don't understand them.

Reply:
there's a `sequential?` check somewhere deeper, where we have a set of emails here. It is technically a sequential field as you may want a certain person first in the `to` field. Here, we don't care.